File: golang/core/easy/convert_1D_array_into_2D/convert_1D_array_into_2D.go
package main

import (
	"fmt"
)

func construct2DArray(original []int, m int, n int) [][]int {
	if (len(original) != m * n) {
		return [][]int{}
	}
	finalArray := make([][]int, m)
	countArray := 0
	for i := 0; i < m; i++ {
		finalArray[i] = make([]int, n)
		for j := 0; j < n; j++ {
			finalArray[i][j] = original[countArray + j]
		}
		countArray += n
	}
  return finalArray
}

func main() {
	fmt.Println(construct2DArray([]int{1, 2, 3, 4}, 2, 2))
}

File: typescript/core/easy/convert_1D_array_into_2D.ts
function construct2DArray(original: number[], m: number, n: number): number[][] {
  if (original.length !== m * n) return []
  let finalArray: number[][] = []
  let countArray = 0
  for (let i = 0; i < m; i++) {
    finalArray[i] = []
    for (let j = 0; j < n; j++) {
      finalArray[i][j] = original[countArray + j]
    }
    countArray += n
  }
  return finalArray
}

console.log(construct2DArray([1,1,1,1], 4, 1))
